pkg/middleend/middleend.go

```diff
@@ -111,7 +118,7 @@ func verifyCreateEncryptedVolumeRequestArgs(in *pb.CreateEncryptedVolumeRequest)
 	case in.EncryptedVolume == nil:
 		log.Println("encrypted_volume should be specified")
 		return errMissingArgument
-	case in.EncryptedVolume.EncryptedVolumeId == nil || in.EncryptedVolume.EncryptedVolumeId.Value == "":
+	case in.EncryptedVolumeId == "":
```

Comment:
`EncryptedVolumeId` is an optional field, if not provided by user, server will fill it in. for example with `uuid` as you showed before

Reply:
Certainly. My oversight. Thank you for the thorough review, Boris!